crates/pathfinder/src/sync/p2p.rs

```diff
@@ -270,6 +278,60 @@ impl Sync {
             }
         }
     }
+
+    async fn sync_state_updates(&self, stop: BlockNumber) -> anyhow::Result<()> {
+        let storage = self.storage.clone();
+        let getter = move |start: BlockNumber,
+                           limit: NonZeroUsize|
+              -> anyhow::Result<Option<SmallVec<[StateUpdateStats; 10]>>> {
+            let mut db = storage
+                .connection()
+                .context("Creating database connection")?;
+            let db = db.transaction().context("Creating database transaction")?;
+            let stats = db
+                .state_update_stats(start.into(), limit)
+                .context("Querying state updates")?;
+            Ok(stats)
+        };
+        let getter = Arc::new(getter);
+
+        while let Some(start) = state_updates::next_missing(self.storage.clone(), stop)
+            .await
+            .context("Finding next missing state update")?
+        {
+            let getter = getter.clone();
+            let result = self
+                .p2p
+                .clone()
+                .contract_updates_stream(start, stop, getter)
+                .map_err(Into::into)
+                .try_chunks(100)
```

Comment:
The aim was to do verification and persistence in larger chunks to utilize cpu and io better (but it's a wild guess rn).

Reply:
I think verification doesn't require ingesting vecors to be concurrent - can just spawn a rayon scope / task or something I think. The vector just "simplifies" construction by allowing for `par_iter`, but also requires copying all the data around again. However, currently you cannot `par_iter` because of the state trie code.

Reply:
Rn chunking is performed just before persisting to persist in bulk.